core/src/main/scala/za/co/absa/spline/harvester/builder/plan/MergeIntoNodeBuilder.scala

```diff
@@ -16,56 +16,94 @@
 
 package za.co.absa.spline.harvester.builder.plan
 
+import org.apache.spark.sql.catalyst.expressions.{Expression => SparkExpression}
 import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 import za.co.absa.commons.reflect.ReflectionUtils.extractValue
 import za.co.absa.spline.harvester.IdGeneratorsBundle
 import za.co.absa.spline.harvester.ModelConstants.CommonExtras
+import za.co.absa.spline.harvester.builder.plan.MergeIntoNodeBuilder._
 import za.co.absa.spline.harvester.converter.{DataConverter, DataTypeConverter}
 import za.co.absa.spline.harvester.postprocessing.PostProcessor
-import za.co.absa.spline.producer.model.{AttrRef, Attribute, DataOperation, FunctionalExpression}
+import za.co.absa.spline.producer.model._
 
 class MergeIntoNodeBuilder
-(override val logicalPlan: LogicalPlan)
+  (logicalPlan: LogicalPlan)
   (idGenerators: IdGeneratorsBundle, dataTypeConverter: DataTypeConverter, dataConverter: DataConverter, postProcessor: PostProcessor)
   extends GenericPlanNodeBuilder(logicalPlan)(idGenerators, dataTypeConverter, dataConverter, postProcessor) {
 
-  private lazy val mergeInputs: Seq[Seq[Attribute]] = inputAttributes.transpose
-
   override lazy val functionalExpressions: Seq[FunctionalExpression] = Seq.empty
 
-  override lazy val outputAttributes: Seq[Attribute] =
-    mergeInputs.map(constructMergeAttribute)
-
-  private def constructMergeAttribute(attributes: Seq[Attribute]) = {
-    val attr1 = attributes.head
-    val idRefs = attributes.map(a => AttrRef(a.id))
-    Attribute(
-      id = idGenerators.attributeIdGenerator.nextId(),
-      dataType = attr1.dataType,
-      childRefs = idRefs,
-      extra = Map(CommonExtras.Synthetic -> true),
-      name = attr1.name
-    )
+  override lazy val outputAttributes: Seq[Attribute] = {
```

Comment:
Will this method work when the input attributes are synthetic? I think in this case the proper way is to take `inputAttributes` and use their id's because they may differ from the Spark ones.